src/main/java/org/traccar/protocol/Xexun2ProtocolDecoder.java

```diff
@@ -176,7 +176,19 @@ protected Object decode(
                         position.set(Position.KEY_SATELLITES, buf.readUnsignedByte());
                         position.setLongitude(convertCoordinate(buf.readDouble()));
                         position.setLatitude(convertCoordinate(buf.readDouble()));
-
+                    }
+                    if (BitUtil.check(positionMask, 7)) {
+                        buf.skipBytes(2); // length = 30
+                        buf.skipBytes(1); // marker = 'G'
+                        buf.skipBytes(2); // length = 27
+                        position.setLongitude(convertCoordinate(buf.readDouble()));
+                        position.setLatitude(convertCoordinate(buf.readDouble()));
+                        position.setValid(buf.readUnsignedByte() > 0); // 0=invalid,1=normal,2=sub-meter,3=differential
+                        position.set(Position.KEY_SATELLITES, buf.readUnsignedByte());
+                        buf.skipBytes(1); // SNR
```

Comment:
This should be `readUnsignedByte` and either remove the comment or change it to full `satellite signal to noise ratio`.